components/raftstore/src/store/worker/pd.rs

```diff
@@ -778,14 +778,7 @@ where
             topn_report.push(read_stat);
         }
 
-        let report_vec = topn_report.into_vec();
-
-        #[cfg(feature = "failpoints")]
-        if report_vec.len() > cap {
-            fail_point!("on_report_vec_too_large");
-        }
```

Comment:
We don't need this, this can never happen.
Capacity is private field and cannot be changed outside, and all public API won't change it, so after you set it in `new`, it can never chage.
If you really want to check it, write in **unit** test instead of in intergration tests.